packages/interactivity/src/directives.js

```diff
@@ -330,7 +330,15 @@ export default () => {
 	// data-wp-text
 	directive( 'text', ( { directives: { text }, element, evaluate } ) => {
 		const entry = text.find( ( { suffix } ) => suffix === 'default' );
-		element.props.children = evaluate( entry );
+		try {
+			if ( typeof evaluate( entry ) === 'object' ) {
+				element.props.children = null;
+				return;
+			}
+			element.props.children = evaluate( entry ).toString();
+		} catch ( e ) {
+			element.props.children = null;
+		}
```

Comment:
It'd be better not to call `evaluate( entry )` twice for each render. People might not be expecting that.

Reply:
What about this:

```suggestion
		try {
			const result = evaluate( entry );
			element.props.children = typeof result === 'object' ? null : result.toString();
		} catch ( e ) {
			element.props.children = null;
		}
```

Reply:
Done in https://github.com/WordPress/gutenberg/pull/57879/commits/ba91a0c362ebeebd9d523d312b6e510e5db35b3c